### PR TITLE
Demerge Student Complete Saga is initially implemented.

### PR DESCRIPTION
**Pull request by infstar**

### CODE REVIEW

File: api/src/main/java/ca/bc/gov/educ/api/pen/services/orchestrator/StudentDemergeCompleteOrchestrator.java [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+    StudentSagaData studentDataFromEventResponse = JsonUtil.getJsonObjectFromString(StudentSagaData.class, event.getEventPayload());
+//    studentDataFromEventResponse.setLegalFirstName(studentDemergeCompleteSagaData.getLegalFirstName());
+//    studentDataFromEventResponse.setLegalLastName(studentDemergeCompleteSagaData.getLegalLastName());
+//    studentDataFromEventResponse.setLegalMiddleNames(studentDemergeCompleteSagaData.getLegalMiddleNames());
+//    studentDataFromEventResponse.setUsualFirstName(studentDemergeCompleteSagaData.getUsualFirstName());
+//    studentDataFromEventResponse.setUsualLastName(studentDemergeCompleteSagaData.getUsualLastName());
+//    studentDataFromEventResponse.setUsualMiddleNames(studentDemergeCompleteSagaData.getUsualMiddleNames());
+//    studentDataFromEventResponse.setDob(studentDemergeCompleteSagaData.getDob());
+//    studentDataFromEventResponse.setGenderCode(studentDemergeCompleteSagaData.getGenderCode());
```
Comment (mishraomp) [inline api/src/main/java/ca/bc/gov/educ/api/pen/services/orchestrator/StudentDemergeCompleteOrchestrator.java:196]: can we delete these commented lines

File: api/src/main/java/ca/bc/gov/educ/api/pen/services/controller/v1/PenServicesSagaController.java [hunk 2]
```diff
@@ -32,12 +35,16 @@
...
   private final StudentMergeCompleteOrchestrator studentMergeCompleteOrchestrator;
 
+  @Getter(PRIVATE)
+  private final StudentDemergeCompleteOrchestrator studentDemergeCompleteOrchestrator;
+
   private static final SagaMapper sagaMapper = SagaMapper.mapper;
 
   @Autowired
-  public PenServicesSagaController(final SagaService sagaService, StudentMergeCompleteOrchestrator studentMergeCompleteOrchestrator) {
+  public PenServicesSagaController(final SagaService sagaService, StudentMergeCompleteOrchestrator studentMergeCompleteOrchestrator, StudentDemergeCompleteOrchestrator studentDemergeCompleteOrchestrator) {
```
Comment (mishraomp) [inline api/src/main/java/ca/bc/gov/educ/api/pen/services/controller/v1/PenServicesSagaController.java:44]: how about injecting List<orchestrator> to the Controller

**Review by mishraomp — CHANGES_REQUESTED** [review]

File: api/src/main/java/ca/bc/gov/educ/api/pen/services/orchestrator/StudentDemergeCompleteOrchestrator.java (hunk 1 above)
Comment (infstar) [reply api/src/main/java/ca/bc/gov/educ/api/pen/services/orchestrator/StudentDemergeCompleteOrchestrator.java]: done

File: api/src/main/java/ca/bc/gov/educ/api/pen/services/controller/v1/PenServicesSagaController.java (hunk 2 above)
Comment (infstar) [reply api/src/main/java/ca/bc/gov/educ/api/pen/services/controller/v1/PenServicesSagaController.java]: done.

**Review by mishraomp — APPROVED** [review]

**Pull request merged**